prow/plugins/trigger/pr.go

```diff
@@ -163,6 +163,13 @@ func buildAll(c client, pr github.PullRequest) error {
 				}
 			}
 			if !job.RunsAgainstChanges(changes) {
+				if err := c.GitHubClient.CreateStatus(org, repo, pr.Head.SHA, github.Status{
```

Comment:
maybe set `skipped = true` here, and change the check below to `if skipped || !job.RunsAgainstBranch...`

Reply:
I was thinking about refactoring out the copied code, but couldn't see a good solution. That looks like a good solution.